controllers/metallb_controller.go

```diff
@@ -70,7 +73,15 @@ func (r *MetallbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
-	logger := r.Log.WithValues("metallb reconcile", req.NamespacedName)
+	if req.Name != defaultMetallbCrName {
```

Comment:
Not sure it's worth the effort but:
- if the bad crd is not deleted, we don't bring the status back
- what happens if the bad crd is there and then we add a good one? Will the degraded condition be removed?

I think it would make sense to cover those cases and add tests for them. What you can do here is to list all metallb resources, check if 1+ bad names exists and then set degraded, instead of looking at the current one being added.


WDYT @mmirecki ?

Reply:
> Not sure it's worth the effort but:
> 
> * if the bad crd is not deleted, we don't bring the status back
> * what happens if the bad crd is there and then we add a good one? Will the degraded condition be removed?
> 

The status is per metallb crd, so each resource gets its own status. 
if we have two - the incorrectly named one would have a degraded condition, the correct one an available one.
When we delete the incorrectly named one, we delete it along with it's status, the status on the correctly named one remains as it was.



> I think it would make sense to cover those cases and add tests for them. What you can do here is to list all metallb resources, check if 1+ bad names exists and then set degraded, instead of looking at the current one being added.
> 
> WDYT @mmirecki ?